src/System.Net.Primitives/tests/PalTests/SocketAddressPalTest.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Sockets;
+
+using Xunit;
+
+namespace System.Net.Primitives.PalTests
+{
+    public class SocketAddressPalTests
+    {
+        public static object[][] AddressFamilyData = new object[][] {
+            new object[] { AddressFamily.Unspecified },
+            new object[] { AddressFamily.Unix },
+            new object[] { AddressFamily.InterNetwork },
+            new object[] { AddressFamily.InterNetworkV6 }
+        };
+
+        [Theory, MemberData("AddressFamilyData")]
+        public void AddressFamily_Get_Set(AddressFamily family)
+        {
+            var buffer = new byte[16];
+            SocketAddressPal.SetAddressFamily(buffer, family);
+            Assert.Equal(family, SocketAddressPal.GetAddressFamily(buffer));
+        }
+
+        [Fact]
+        public void AddressFamily_Get_Set_Throws()
+        {
+            var buffer = new byte[1];
+            Assert.Throws<Exception>(() => SocketAddressPal.SetAddressFamily(buffer, AddressFamily.InterNetwork));
+            Assert.Throws<Exception>(() => SocketAddressPal.GetAddressFamily(buffer));
+        }
+
```

Comment:
Looks like one of these asserts failed in CI:
```
System.Net.Primitives.PalTests.SocketAddressPalTests.AddressFamily_Get_Set_Throws [FAIL]
        Assert.Throws() Failure
        Expected: typeof(System.Exception)
        Actual:   typeof(System.IndexOutOfRangeException): Index was outside the bounds of the array.
        Stack Trace:
d:\j\workspace\dotnet_corefx_windows_debug_prtest\packages\Microsoft.DotNet.BuildTools.1.0.25-prerelease-00066\lib\tests.targets(130,5): warning MSB3073: The command "CoreRun.exe xunit.console.netcore.exe System.Net.Primitives.Windows.Pal.Tests.dll  -xml testResults.xml -notrait category=nonwindowstests  -notrait category=OuterLoop -notrait category=failing " exited with code -2146232797. [d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Net.Primitives\tests\PalTests\System.Net.Primitives.Windows.Pal.Tests.csproj]
d:\j\workspace\dotnet_corefx_windows_debug_prtest\packages\Microsoft.DotNet.BuildTools.1.0.25-prerelease-00066\lib\tests.targets(141,5): error : One or more tests failed while running tests from 'System.Net.Primitives.Windows.Pal.Tests' please check log for details! [d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Net.Primitives\tests\PalTests\System.Net.Primitives.Windows.Pal.Tests.csproj]
```
Presumably either one or both should be changed to ```Throws<IndexOutOfRangeException>``` or to ```ThrowsAny<Exception>```.